packages/rum/src/performanceCollection.ts

```diff
@@ -99,49 +101,58 @@ export function startPerformanceCollection(lifeCycle: LifeCycle) {
   }
 }
 
-export function retrieveInitialDocumentResourceTiming() {
-  let timing: RumPerformanceResourceTiming
+export function retrieveInitialDocumentResourceTiming(callback: (timing: RumPerformanceResourceTiming) => void) {
+  runOnReadyState('interactive', () => {
+    let timing: RumPerformanceResourceTiming
 
-  const forcedAttributes = {
-    entryType: 'resource' as const,
-    initiatorType: FAKE_INITIAL_DOCUMENT,
-    traceId: getDocumentTraceId(document),
-  }
-  if (supportPerformanceNavigationTimingEvent() && performance.getEntriesByType('navigation').length > 0) {
-    const navigationEntry = performance.getEntriesByType('navigation')[0]
-    timing = { ...navigationEntry.toJSON(), ...forcedAttributes }
-  } else {
-    const relativePerformanceTiming = computeRelativePerformanceTiming()
-    timing = {
-      ...relativePerformanceTiming,
-      decodedBodySize: 0,
-      duration: relativePerformanceTiming.responseEnd,
-      name: window.location.href,
-      startTime: 0,
-      ...forcedAttributes,
+    const forcedAttributes = {
+      entryType: 'resource' as const,
+      initiatorType: FAKE_INITIAL_DOCUMENT,
+      traceId: getDocumentTraceId(document),
     }
-  }
-  return timing
+    if (supportPerformanceNavigationTimingEvent() && performance.getEntriesByType('navigation').length > 0) {
+      const navigationEntry = performance.getEntriesByType('navigation')[0]
+      timing = { ...navigationEntry.toJSON(), ...forcedAttributes }
+    } else {
+      const relativePerformanceTiming = computeRelativePerformanceTiming()
+      timing = {
+        ...relativePerformanceTiming,
+        decodedBodySize: 0,
+        duration: relativePerformanceTiming.responseEnd,
+        name: window.location.href,
+        startTime: 0,
+        ...forcedAttributes,
+      }
+    }
+    callback(timing)
+  })
 }
 
 function retrieveNavigationTimingWhenLoaded(callback: (timing: RumPerformanceNavigationTiming) => void) {
-  function sendFakeTiming() {
-    callback({
-      ...computeRelativePerformanceTiming(),
-      entryType: 'navigation',
-    })
-  }
+  runOnReadyState('complete', () => {
+    // Send it a bit after the actual load event, so the "loadEventEnd" timing is accurate
+    setTimeout(
+      monitor(() => {
+        callback({
```

Comment:
Good to go for me. 
I just think that having the named function sendFakeTiming was quite useful for the understanding of this code block.
```
function sendFakeTiming() {
    callback({
      ...computeRelativePerformanceTiming(),
      entryType: 'navigation',
    })
  }
```
I would be un favor of keeping it :


```
runOnReadyState('complete', () => {
    // Send it a bit after the actual load event, so the "loadEventEnd" timing is accurate
    setTimeout(monitor(sendFakeTiming))
})
```